dev/tests/api-functional/testsuite/Magento/GraphQl/CatalogCustomer/PriceTiersTest.php

```diff
@@ -117,6 +117,40 @@ public function testGetLowestPriceForGuest()
         $this->assertEquals(round(7.25, 2), $this->getValueForQuantity(8, $itemTiers));
     }
 
+    /**
+     * @magentoApiDataFixture Magento/Catalog/_files/simple_product_without_tier_price.php
+     * @magentoApiDataFixture Magento/Catalog/_files/three_simple_products_with_tier_price.php
+     */
+    public function testGetCorrectDisplaingTierPriceForProducts()
+    {
+        $productSku = 'simple';
+        $query =  <<<QUERY
+{
+  products(search: "{$productSku}") {
+   items {
+        sku
+        name
+          price_tiers {
+              quantity
+              final_price {
+                  value
+              }
+          }
+      }
+  }
+}
+QUERY;
+        $response = $this->graphQlQuery($query);
+        foreach ($response['products']['items'] as $key => $item) {
+            if ($item['sku'] == 'simple_19') {
+                $this->assertCount(0, $response['products']['items'][$key]['price_tiers']);
+            }
+            if (in_array($item['sku'], ['simple_1','simple_2','simple_3'])) {
+                $this->assertCount(1, $response['products']['items'][$key]['price_tiers']);
+            }
```

Comment:
For a better code readability, I would refactor it this way. In case you have a better and cleaner way, feel free to implement it.

```suggestion
            $productsWithTierPrices = ['simple_1','simple_2','simple_3'];
            
            if (in_array($item['sku'], $productsWithTierPrices)) {
                $this->assertCount(1, $response['products']['items'][$key]['price_tiers']);
            } else {
                $this->assertCount(0, $response['products']['items'][$key]['price_tiers']);
            }
```